discord/ext/commands/bot.py

```diff
@@ -129,10 +148,10 @@ def __init__(self, command_prefix, help_command=_default, description=None, **op
     # internal helpers
 
     def dispatch(self, event_name, *args, **kwargs):
-        super().dispatch(event_name, *args, **kwargs)
+        super().dispatch(event_name, *args, **kwargs)  # type: ignore
```

Comment:
any attribute of `Client` accessed in `BotBase` will need to be type ignored. unless we do
`assert isinstance(self, Client)` at the top of each call or have BotBase subclass Client.

Reply:
I believe you should be able to set `self: Client` in the method signature

Reply:
![image](https://user-images.githubusercontent.com/8677174/129353437-ff66280b-9ef5-4609-9d79-4eb45fd5f9f6.png)
seems at least with pyright this is not the case.

Reply:
Ok, I must have misread something about advanced uses of `self` types. I'll see if I can figure out a way to do this right.